cpu/cpu.go

```diff
@@ -182,26 +196,43 @@ func (p *Plugin) CollectMetrics(metricTypes []plugin.MetricType) ([]plugin.Metri
 		p.snapMetricsNames, p.procStatMetricsNames); err != nil {
 		return nil, err
 	}
+	ts := time.Now()
+	prefix := filepath.Join(vendor, fs, pluginName)
 	for _, metricType := range metricTypes {
 		ns := metricType.Namespace()
 		if len(ns) != maxNamespaceSize {
 			return nil, fmt.Errorf("Incorrect namespace length (len = %d)", len(ns))
 		}
-
-		val, err := getMapValueByNamespace(p.stats[ns.Strings()[3]], ns.Strings()[4:])
-
-		if err != nil {
-			return metrics, err
-		}
-
-		metric := plugin.MetricType{
-			Namespace_: ns,
-			Data_:      val,
-			Timestamp_: time.Now(),
+		if ns[len(ns)-2].Value == "*" {
+			for cpuId, cpuStats := range p.stats {
+				for k, v := range cpuStats {
+					if ns[len(ns)-1].Value == k && v != nil {
+						ns1 := core.NewNamespace(strings.Split(prefix, string(os.PathSeparator))...).
+							AddStaticElement(cpuId).
```

Comment:
@IzabellaRaulin: good catch!  

@obourdon: If an element in the namespace is returned from GetMetricTypes (populating the catalog) as dynamic we should, in general, exit from CollectMetrics with it also being dynamic (though it should have the value set as @IzabellaRaulin pointed out).  A primary reason for this is to enable downstream plugins to be able to parse the namespace for data given enough context to be useful.  You can see an example of using dynamic elements of the name space in a publisher [here](https://github.com/intelsdi-x/snap-plugin-publisher-influxdb/blob/master/influx/influx.go#L163-L176).